### ISSUE TITLE
allow-any-import-level seems to have no effect

### ISSUE THREAD

**Issue by paugier** [pylint-dev/pylint#9456]
### Bug description

```python
I try to use `allow-any-import-level` but it does not seem to have any effect.

With a module with:


MY_CONST = 1
import numpy
print(numpy)


`pylint mod.py --allow-any-import-level numpy` produces the error message:

```
mod.py:7:0: C0413: Import "import numpy" should be placed at the top of the module (wrong-import-position)
```

I tried with a configuration file and with different variations of the command but I didn't manage to correctly use `allow-any-import-level`.
```

### Command used

```shell
pylint mod.py --allow-any-import-level numpy
```

### Pylint output

```python
mod.py:7:0: C0413: Import "import numpy" should be placed at the top of the module (wrong-import-position)
```

### Expected behavior

No message C0413 for numpy imports.

### Pylint version

```shell
pylint 3.0.3
astroid 3.0.3
Python 3.9.2 (default, Feb 28 2021, 17:03:44) 
[GCC 10.2.1 20210110]
```

### OS / Environment

Linux

**Comment by mbyrnepr2** [comment]
Thanks @paugier!

[allow-any-import-level](https://pylint.readthedocs.io/en/stable/user_guide/configuration/all-options.html#allow-any-import-level) doesn't supress [wrong-import-position](https://pylint.readthedocs.io/en/stable/user_guide/messages/convention/wrong-import-position.html); it suppresses [import-outside-toplevel](https://pylint.readthedocs.io/en/stable/user_guide/messages/convention/import-outside-toplevel.html).

Is there a way we can improve the situation via the documentation?
